src/components/memorygame/shuffle.jsx

```diff
@@ -0,0 +1,11 @@
+// Check Fisher–Yates shuffle algorithm https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
+const shuffle = (arr) => {
```

Comment:
nondestructive shuffle goes here; think about whether it makes more sense to comment "non-destructive shuffle" above every call to this function or in the place the function is declared (if i used htis 100x in my codebase and wrote it above every time, that would seem a bit tedious)

Reply:
fixed